courseworks/week6/ZhigaoXIE/Q2.jl

```diff
@@ -0,0 +1,27 @@
+# Q2.jl
+
+using SparseArrays, Test
+using LinearAlgebra
+
+function my_spv(A::SparseMatrixCSC{T1}, x::AbstractVector{T2}) where {T1, T2}
+    T = promote_type(T1, T2)
+    @assert size(A, 2) == length(x)
+    y = spzeros(T, size(A, 1))
```

Comment:
As we mentioned in the class, it not a wise idea to used $n \times 1$ sparse matrix to represent a 1d array because it is actually full.